activerecord/lib/active_record/table_metadata.rb

```diff
@@ -19,7 +19,7 @@ def type(column_name)
     end
 
     def has_column?(column_name)
-      klass&.columns_hash.key?(column_name)
+      klass&.columns_hash&.key?(column_name)
```

Comment:
```suggestion
      klass.columns_hash.key?(column_name) if klass
```

Would perhaps be more readable, but no big deal.